pkgs/applications/science/electronics/alliance/default.nix

```diff
@@ -1,74 +1,46 @@
-{ lib, stdenv, fetchurl
-, xorgproto, motif, libX11, libXt, libXpm, bison
-, flex, automake, autoconf, libtool, runtimeShell
+{ lib, stdenv, fetchFromGitLab, xorgproto, motif, libX11, libXt, libXpm, bison
+, flex, automake, autoconf, libtool
 }:
 
 stdenv.mkDerivation rec {
   pname = "alliance";
-  version = "5.1.1";
-
-  src = fetchurl {
-    url = "http://www-asim.lip6.fr/pub/alliance/distribution/5.0/${pname}-${version}.tar.bz2";
-    sha256 = "046c9qwl1vbww0ljm4xyxf5jpz9nq62b2q0wdz9xjimgh4c207w1";
+  version = "unstable-2020-12-04";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.lip6.fr";
+    owner = "vlsi-eda";
+    repo = "alliance";
+    rev = "c12f377989571c1d868bebebdfb1574572e2ccf9";
+    sha256 = "RNfW621hRx5pzg5GhtY4sG8jwuQws3oG8EjZF7Mt5B0=";
   };
 
+  prePatch = "cd alliance/src";
 
   nativeBuildInputs = [ libtool automake autoconf flex ];
   buildInputs = [ xorgproto motif libX11 libXt libXpm bison ];
 
-  sourceRoot = "alliance/src/";
+  ALLIANCE_TOP = placeholder "out";
 
   configureFlags = [
-    "--prefix=$(out)"
+    "--prefix=${placeholder "out"}" "--enable-alc-shared"
   ];
 
-  preConfigure = ''
-    mkdir -p $out/etc
-
-    #texlive for docs seems extreme
-    mkdir -p $out/share/alliance
-    mv ./documentation $out/share/alliance
+  postPatch = ''
+    # texlive for docs seems extreme
     substituteInPlace autostuff \
-      --replace "$newdirs documentation" "$newdirs" \
-      --replace documentation Solaris
-
-    substituteInPlace sea/src/DEF_grammar_lex.l \
-      --replace "ifndef FLEX_BETA" "if (YY_FLEX_MAJOR_VERSION <= 2) && (YY_FLEX_MINOR_VERSION < 6)"
-    ./autostuff
-  '';
+      --replace "$newdirs documentation" "$newdirs"
```

Comment:
this seems to be fixed on master, maybe make an update?

Reply:
I am not sure what you mean by this.

Reply:
this is patched upstream master https://gitlab.lip6.fr/vlsi-eda/alliance/-/commit/720e37e77c3dd04e91345c232bfb12d00b1d1539

Reply:
Ah, I didn't realize there were new commits! Unexpected.

Reply:
The new commits seems to have some issues, I get errors that are unrelated to Nix and related to the code being wrong. I don't think we can use the new commits.

Reply:
well thats sad

Reply:
Nevermind, it's because I had parallel building enabled.

Reply:
I couldn't use the latest commit anyway, it had some odd linking error I couldn't figure out, and considering that the commit's purpose was only to fix a linking error the author had, I just used the second latest commit.